ansible_ai_connect/ai/api/views.py

```diff
@@ -1012,3 +1016,77 @@ def write_to_segment(
                 },
             )
         send_segment_event(event, "codegenPlaybook", user)
+
+
+class Chat(APIView):
+    """
+    Send a message to the backend chatbot service and get a reply.
+    """
+
+    permission_classes = [
+        permissions.IsAuthenticated,
+        IsAuthenticatedOrTokenHasScope,
+    ]
+    required_scopes = ["read", "write"]
+
+    CHAT_API_NOT_ENABLED = "Chat API is not enabled."
+    INVALID_CHAT_REQUEST = "An invalid chat request was received."
+    INVALID_CHAT_RESPONSE = "An invalid chat response was received."
+
+    def __init__(self):
+        self.chat_enabled = (
+            settings.CHATBOT_URL
+            and settings.CHATBOT_DEFAULT_MODEL
+            and settings.CHATBOT_DEFAULT_PROVIDER
+        )
+
+    @extend_schema(
+        request=ChatRequestSerializer,
+        responses={
+            200: ChatResponseSerializer,
+            400: OpenApiResponse(description="Bad Request"),
+        },
+        summary="Chat request",
+    )
+    def post(self, request) -> Response:
```

Comment:
I'm not one to argue... I like `POST` but @jameswnl eluded to "chat" APIs using a Web socket to keep a session alive. Happy with `POST` I'm just curious.

Reply:
In this version, this is implemented basically a proxy to OpenShift lightspeed-service's `POST /v1/query` API minus attachment support. We may want to use a web socket later... but not now.